docs/concept/#894-Efficient-orchestration-of-edc-negotiations/#894-Efficient-orchestration-of-edc-negotiations.md

```diff
@@ -0,0 +1,103 @@
+|---------------|--------------------------------------------------------------------------|
+| Creation date | 09.10.2024                                                               |
+| Ticket Id     | https://github.com/eclipse-tractusx/item-relationship-service/issues/894 |    
+| State         | DONE                                                                     | 
+
+## Table of Contents
+
+1. [Decision](#Decision)
+2. [Rationale](#Rationale)
+3. [Approach](#Approach)
+4. [Testing Strategy](#Testing-Strategy)
+
+## Decision
+
+Implementation of a new EDC Orchestrator within the EDC Client Library for efficient management of negotiations. The
+goal is to prevent duplicate negotiations for assets, thereby ensuring more efficient processing in asset consumption.
+
+## Rationale
+
+Each job is executed in isolation. This results in many jobs conducting a multitude of unnecessary contract negotiations
+for the same DT Registry Asset in the EDC catalog. In case the same catalog offer is requested in different jobs this
+leads to not performant overall executions. Therefore, a concept is required how to handle this in the most efficient
+way.
+
+## Approach
+
+A new service “NegotiationOrchestrator” is introduced which is a middle layer between the already existing components
+EdcSubmodelClientImpl and ContractNegotiationService, EDCCatalogFacade and EndpointDataReferenceCacheService.
+
+The service handles the following tasks:
+
+- orchestration of catalog requests by id
+- orchestration of catalog requests by type
+- orchestration of contract negotiations
+- caching of ongoing contract negotiations
+- caching of EDR tokens
+
+To prevent too many parallel requests to the EDC, an ExecutorService handles the catalog requests and negotiations with
+a defined thread pool.
+
+To cache the currently ongoing negotiations, a ConcurrentHashMap with EDC AssetIds as key can be used. Due to the
+asynchronous process within NegotiationOrchestrator, the future EDR can be created and added to the cache as soon as the
+method is called and will be removed only after the negotiation is completed. In addition, a time to live can be added
+for cleanup.
+
+To cache EDR tokens, the EndpointDataReferenceCacheService can be used. The caching service wraps EDR tokens into an
+object with a state indicating if the token is valid (i.e. can be used directly to authenticate the dataplane request)
+or expired (i.e. the validityPeriod is expired). After expiration the token can be reused to request a new token. Cache
+eviction is handled independent of token state and will use a configurable TTL to evict cached entries.
+
+![edc-flow-orchestrated.svg](edc-flow-orchestrated.svg)
+
+See [docs/src/uml-diagrams/runtime-view/edc-flow-orchestrated.puml](https://github.com/eclipse-tractusx/item-relationship-service/blob/main/docs/src/uml-diagrams/runtime-view/edc-flow-orchestrated.puml)
```

Comment:
Usage of a table description where you define the remarks according to the numbers in the diagramm

Reply:
all steps should be described in more detail.